### PR TITLE
Let rust-analyzer ship on stable, non-preview

### PR DESCRIPTION
**Pull request by cuviper**
The consensus on rust-lang/rust-analyzer#12432 seems to be that we are ready for `rust-analyzer` to ship as a rustup component on the beta and stable channels. This won't always be the preferred distribution method, e.g. the VS Code extension will probably still independently update to its weekly releases, but it's still useful to have a component that follows the release train with the rest of the Rust toolchain. So this removes the nightly-only gating on the bundled component, and removes the "-preview" suffix as well by the usual renaming mechanism.

cc @rust-lang/wg-rls-2 @rust-lang/release

### CODE REVIEW

**Review by lnicola — APPROVED** [review]

File: src/bootstrap/dist.rs [hunk 1]
```diff
@@ -1044,12 +1044,6 @@ impl Step for RustAnalyzer {
...
 
     fn run(self, builder: &Builder<'_>) -> Option<GeneratedTarball> {
-        // This prevents rust-analyzer from being built for "dist" or "install"
-        // on the stable/beta channels. It is a nightly-only tool and should
-        // not be included.
-        if !builder.build.unstable_features() {
-            return None;
-        }
         let compiler = self.compiler;
         let target = self.target;
```
Comment (Mark-Simulacrum) [inline src/bootstrap/dist.rs:1049]: There's a tarball.is_preview below that needs updating I think.

Comment (cuviper) [reply src/bootstrap/dist.rs:1049]: I don't think so, but cause the actual component remains "*-preview" for the purpose of rustup updates, while the new rename is basically just an alias. We have the same thing still set on all formerly-preview components, even rustfmt.

Comment (Mark-Simulacrum) [reply src/bootstrap/dist.rs:1049]: Ah, interesting. We should consider if there's a way to migrate, feels off that we need to keep what seems like the canonical name the same. Maybe the renames need to be in the other direction or something too?

**Comment by Mark-Simulacrum** [comment]
It looks like src/bootstrap/tool.rs needs an update to RustAnalyzer to set stable=true too.

r=me with that done.

File: src/bootstrap/dist.rs (hunk 1 above)
Comment (cuviper) [reply src/bootstrap/dist.rs:1049]: Maybe so, but we'll need to work with rustup folks to be sure the upgrade path works.

**Pull request merged**